src/server/server_family.cc

```diff
@@ -2064,6 +2149,28 @@ void ServerFamily::ReplicaOf(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void ServerFamily::ReplicaOf(CmdArgList args, ConnectionContext* cntx) {
+  string_view host = ArgS(args, 0);
+  string_view port = ArgS(args, 1);
+
+  ReplicaOfInternal(host, port, cntx, ShouldFlushDb::kFlush,
```

Comment:
Can't we first flush the DB and then call `ReplicaOfInternal()` here? Does it have to be inside?
By doing it here, we'll reduce the complexity of `ReplicaOfInternal()` and also remove a parameter (which to be is somewhat odd to have for this function)